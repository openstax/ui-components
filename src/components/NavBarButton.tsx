import { Button, ButtonProps } from "react-aria-components";
import styled from "styled-components";

export type NavBarButtonProps = Omit<ButtonProps, "aria-label"> & {
  label?: string;
  icon?: string | React.ReactNode;
  className?: string;
  "aria-label"?: string;
} & ({ label: string } | { "aria-label": string });

export const NavBarButton = styled(
  ({
    label,
    icon,
    className,
    "aria-label": ariaLabel,
    ...props
  }: NavBarButtonProps) => (
    <Button className={className} {...props}>
      {icon &&
        (typeof icon === "string" ? (
          <img aria-hidden="true" src={icon} alt="" />
        ) : (
          icon
        ))}
      {label ? <span>{label}</span> : null}
    </Button>
  ),
)`
  border: none;
  background: none;
  padding: 0;
  min-height: 4rem;
  min-width: 4rem;
  height: 100%;
  display: inline-flex;
  place-content: center;
  align-items: center;
  cursor: pointer;
  font-weight: 500;

  img {
    max-height: 100%;
  }

  img + *,
  svg + * {
    margin-left: 0.8rem;
  }
`;
