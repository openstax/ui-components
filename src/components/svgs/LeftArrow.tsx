export const LeftArrow = (props: React.SVGProps<SVGSVGElement>) => (
  <svg
    width="8"
    height="10"
    viewBox="0 0 8 10"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      d="M1.1266 5.38862L5.57702 9.83902C5.79166 10.0537 6.13964 10.0537 6.35426 9.83902L6.87333 9.31995C7.0876 9.10568 7.08801 8.75841 6.87424 8.54363L3.34721 4.99999L6.87424 1.45637C7.08801 1.24159 7.0876 0.89432 6.87333 0.680047L6.35426 0.160979C6.13962 -0.0536598 5.79164 -0.0536598 5.57702 0.160979L1.12662 4.61138C0.911981 4.826 0.911981 5.17398 1.1266 5.38862Z"
      fill="#959595"
    />
  </svg>
);
