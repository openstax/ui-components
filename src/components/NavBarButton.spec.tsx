import renderer from 'react-test-renderer';
import { NavBarButton } from './NavBarButton';
import { Info } from './svgs/Info';

describe('NavBarButton', () => {
  it('matches text snapshot', () => {
    const tree = renderer.create(
      <NavBarButton label="button" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches svg icon snapshot', () => {
    const tree = renderer.create(
      <NavBarButton icon={<Info />} aria-label="info" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches base64 icon snapshot', () => {
    const tree = renderer.create(
      <NavBarButton icon="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAiIGhlaWdodD0iNTYiIHZpZXdCb3g9IjAgMCAxMCA1NiIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8Y2lyY2xlIGN4PSI1IiBjeT0iNSIgcj0iNSIgZmlsbD0iIzAwMCIvPgogIDxjaXJjbGUgY3g9IjUiIGN5PSIyOCIgcj0iNSIgZmlsbD0iIzAwMCIvPgogIDxjaXJjbGUgY3g9IjUiIGN5PSI1MSIgcj0iNSIgZmlsbD0iIzAwMCIvPgo8L3N2Zz4K" aria-label="dots menu" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches icon and text snapshot', () => {
    const tree = renderer.create(
      <NavBarButton label="Info" icon={<Info />} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
