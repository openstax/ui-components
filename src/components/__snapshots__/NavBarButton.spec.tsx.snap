// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarButton matches base64 icon snapshot 1`] = `
<button
  aria-label="dots menu"
  className="sc-bczRLJ dKgZvy"
  data-rac=""
  data-react-aria-pressable={true}
  id="react-aria-5"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <img
    alt=""
    aria-hidden="true"
    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAiIGhlaWdodD0iNTYiIHZpZXdCb3g9IjAgMCAxMCA1NiIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8Y2lyY2xlIGN4PSI1IiBjeT0iNSIgcj0iNSIgZmlsbD0iIzAwMCIvPgogIDxjaXJjbGUgY3g9IjUiIGN5PSIyOCIgcj0iNSIgZmlsbD0iIzAwMCIvPgogIDxjaXJjbGUgY3g9IjUiIGN5PSI1MSIgcj0iNSIgZmlsbD0iIzAwMCIvPgo8L3N2Zz4K"
  />
</button>
`;

exports[`NavBarButton matches icon and text snapshot 1`] = `
<button
  className="sc-bczRLJ dKgZvy"
  data-rac=""
  data-react-aria-pressable={true}
  id="react-aria-7"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <svg
    fill="none"
    height="13"
    viewBox="0 0 12 13"
    width="12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6 0.180664C2.68652 0.180664 0 2.86816 0 6.18066C0 9.49511 2.68652 12.1807 6 12.1807C9.31348 12.1807 12 9.49511 12 6.18066C12 2.86816 9.31348 0.180664 6 0.180664ZM6 2.84195C6.56119 2.84195 7.01613 3.29689 7.01613 3.85808C7.01613 4.41928 6.56119 4.87421 6 4.87421C5.43881 4.87421 4.98387 4.41928 4.98387 3.85808C4.98387 3.29689 5.43881 2.84195 6 2.84195ZM7.35484 8.98712C7.35484 9.14745 7.22485 9.27744 7.06452 9.27744H4.93548C4.77515 9.27744 4.64516 9.14745 4.64516 8.98712V8.40647C4.64516 8.24614 4.77515 8.11615 4.93548 8.11615H5.22581V6.56776H4.93548C4.77515 6.56776 4.64516 6.43777 4.64516 6.27744V5.69679C4.64516 5.53646 4.77515 5.40647 4.93548 5.40647H6.48387C6.6442 5.40647 6.77419 5.53646 6.77419 5.69679V8.11615H7.06452C7.22485 8.11615 7.35484 8.24614 7.35484 8.40647V8.98712Z"
      fill="#959595"
    />
  </svg>
  <span>
    Info
  </span>
</button>
`;

exports[`NavBarButton matches svg icon snapshot 1`] = `
<button
  aria-label="info"
  className="sc-bczRLJ dKgZvy"
  data-rac=""
  data-react-aria-pressable={true}
  id="react-aria-3"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <svg
    fill="none"
    height="13"
    viewBox="0 0 12 13"
    width="12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6 0.180664C2.68652 0.180664 0 2.86816 0 6.18066C0 9.49511 2.68652 12.1807 6 12.1807C9.31348 12.1807 12 9.49511 12 6.18066C12 2.86816 9.31348 0.180664 6 0.180664ZM6 2.84195C6.56119 2.84195 7.01613 3.29689 7.01613 3.85808C7.01613 4.41928 6.56119 4.87421 6 4.87421C5.43881 4.87421 4.98387 4.41928 4.98387 3.85808C4.98387 3.29689 5.43881 2.84195 6 2.84195ZM7.35484 8.98712C7.35484 9.14745 7.22485 9.27744 7.06452 9.27744H4.93548C4.77515 9.27744 4.64516 9.14745 4.64516 8.98712V8.40647C4.64516 8.24614 4.77515 8.11615 4.93548 8.11615H5.22581V6.56776H4.93548C4.77515 6.56776 4.64516 6.43777 4.64516 6.27744V5.69679C4.64516 5.53646 4.77515 5.40647 4.93548 5.40647H6.48387C6.6442 5.40647 6.77419 5.53646 6.77419 5.69679V8.11615H7.06452C7.22485 8.11615 7.35484 8.24614 7.35484 8.40647V8.98712Z"
      fill="#959595"
    />
  </svg>
</button>
`;

exports[`NavBarButton matches text snapshot 1`] = `
<button
  className="sc-bczRLJ dKgZvy"
  data-rac=""
  data-react-aria-pressable={true}
  id="react-aria-1"
  onBlur={[Function]}
  onClick={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span>
    button
  </span>
</button>
`;
